docs/reference/api/pytorch.txt

```diff
@@ -144,6 +146,32 @@ To perform gradient clipping Determined provides two pre-made callback classes:
 .. autoclass:: determined.pytorch.ClipGradsL2Value
     :members:
 
+
+Migration guide
```

Comment:
blocking: Since documentation is long-lived, I'm not certain a migration guide makes sense in this API reference; this documentation will only work for users who are both familiar with the old interface, and want to move over to the new interface. Instead, I think this document should be updated to reflect the current state of the API. However, a migration guide is a great idea: perhaps for the release note, or even a separate article just for folks who do want to upgrade.

Reply:
I see. Will only put it in release notes.

Reply:
According to the dogfooding feedback, the two styles of the API are confusing. I change that by grouping the functions/configurations that users need to specify into two lists and move the migration plan to a note snippet. It might make sense to have a migration guide in the reference because we are supporting two styles of APIs in the same class. However, that note snipper might be a bit redundant.